conda_build/mapdeps.py

```diff
@@ -0,0 +1,107 @@
+"""
+Tools for doing dependency mappings for recipes
+"""
+
+import sys
+import unicodedata
+import yaml
+
+def load(fn):
+    rslt = []
+    try:
+      with open(fn, 'r') as stream:
+        rslt = yaml.safe_load(stream)
+    except:
+      print("Unable to load {} as yaml".format(fn))
+    # get_for_cran(rslt, 'rcpparmadillo')
+    return rslt
+
+def get_sysreqs(deps, sysreqs):
+    rslt = []
+    strs = []
+    for d in sysreqs:
+      try:
+        v = deps['systemreq']
+        for i in v:
+           try:
+             name = i['dep'].lower()
+             if name in d.lower():
+               cran_name = 'SR_' + i['as']
+               if not cran_name in strs:
+                  strs.append(cran_name)
+           except:
+             pass
+      except:
+        pass
+
+    try:
+      v = deps['dependencies']
+      for i in v:
+        try:
+          fn = i['for_cran']
+          for fcn in fn:
+            if fcn in strs:
+              rslt.append(i)
+              break
+        except:
+          pass
+    except:
+      pass
+    return rslt
+
+def get_for_cran(deps, crandep):
```

Comment:
I think you can do the job here just using one list comphreension.
Something like:
```python
def get_for_cran(deps, cran_dep):
    return [i for i in deps.get("dependencies", []) if cran_dep in i.get("for_cran", "")]
```